// firebaseui-angular/esm5/lib/firebaseui-angular-library.helper.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/*
 * Created by Raphael Jenni
 * Copyright (c) 2017 Raphael Jenni
 */
var FirebaseUIAuthConfig = /** @class */ (function () {
    function FirebaseUIAuthConfig() {
    }
    return FirebaseUIAuthConfig;
}());
export { FirebaseUIAuthConfig };
function FirebaseUIAuthConfig_tsickle_Closure_declarations() {
    /** @type {?} */
    FirebaseUIAuthConfig.prototype.providers;
    /** @type {?} */
    FirebaseUIAuthConfig.prototype.method;
    /** @type {?} */
    FirebaseUIAuthConfig.prototype.signInSuccessUrl;
    /** @type {?} */
    FirebaseUIAuthConfig.prototype.tos;
    /** @type {?} */
    FirebaseUIAuthConfig.prototype.credentialHelper;
}
var FirebaseUISignInSuccess = /** @class */ (function () {
    function FirebaseUISignInSuccess() {
    }
    return FirebaseUISignInSuccess;
}());
export { FirebaseUISignInSuccess };
function FirebaseUISignInSuccess_tsickle_Closure_declarations() {
    /** @type {?} */
    FirebaseUISignInSuccess.prototype.currentUser;
    /** @type {?} */
    FirebaseUISignInSuccess.prototype.credential;
    /** @type {?} */
    FirebaseUISignInSuccess.prototype.redirectUrl;
}
/** @enum {number} */
var CredentialHelper = {
    AccountChooser: 0,
    OneTap: 1,
    None: 2,
};
export { CredentialHelper };
CredentialHelper[CredentialHelper.AccountChooser] = "AccountChooser";
CredentialHelper[CredentialHelper.OneTap] = "OneTap";
CredentialHelper[CredentialHelper.None] = "None";
/** @enum {number} */
var AuthProvider = {
    Google: 0,
    Facebook: 1,
    Twitter: 2,
    Github: 3,
    Password: 4,
    Phone: 5,
};
export { AuthProvider };
AuthProvider[AuthProvider.Google] = "Google";
AuthProvider[AuthProvider.Facebook] = "Facebook";
AuthProvider[AuthProvider.Twitter] = "Twitter";
AuthProvider[AuthProvider.Github] = "Github";
AuthProvider[AuthProvider.Password] = "Password";
AuthProvider[AuthProvider.Phone] = "Phone";
/**
 * @record
 */
export function AuthProviderWithCustomConfig() { }
function AuthProviderWithCustomConfig_tsickle_Closure_declarations() {
    /** @type {?} */
    AuthProviderWithCustomConfig.prototype.provider;
    /** @type {?} */
    AuthProviderWithCustomConfig.prototype.customConfig;
}
/** @enum {number} */
var AuthMethods = {
    Popup: 0,
    Redirect: 1,
};
export { AuthMethods };
AuthMethods[AuthMethods.Popup] = "Popup";
AuthMethods[AuthMethods.Redirect] = "Redirect";
